spyderlib/widgets/file_switcher.py

```diff
@@ -0,0 +1,401 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © 2013-2015 The Spyder Development Team
+# Copyright © 2015 Daniel Manson (@d1manson)
+# Licensed under the terms of the MIT License
+# (see spyderlib/__init__.py for details)
+
+from __future__ import print_function
+import os
+import os.path as osp
+
+from spyderlib.qt.QtCore import Signal, Qt, QObject, QSize, QEvent
+from spyderlib.qt.QtGui import (QVBoxLayout, QHBoxLayout,
+                                QListWidget, QListWidgetItem,
+                                QDialog, QLineEdit, QStyledItemDelegate,
+                                QStyleOptionViewItemV4, QApplication,
+                                QTextDocument, QStyle,
+                                QAbstractTextDocumentLayout)
+
+# Local imports
+from spyderlib.baseconfig import _
+from spyderlib.guiconfig import new_shortcut
+from spyderlib.py3compat import to_text_string
+from spyderlib.utils.qthelpers import get_std_icon
+from spyderlib.widgets.helperwidgets import HelperToolButton
+
+
+def shorten_paths(path_list):
+    """
+    Takes a list of paths and tries to "intelligently" shorten them all.  The
+    aim is to make it clear to the user where the paths differ, as that is
+    likely what they care about.  Note that this operates on a list of paths
+    not individual paths.
+
+    If the path ends in an actual file name, it will be trimmed off.
+
+    TODO: at the end, if the path is too long, should do a more dumb kind of
+    shortening, but not completely dumb.
+    """
+
+    # convert the path strings to a list of tokens
+    # and start building the new_path using the drive
+    path_list = path_list[:]  # clone locally
+    new_path_list = []
+    for ii, path in enumerate(path_list):
+        drive, path = osp.splitdrive(osp.dirname(path))
+        new_path_list.append(drive + osp.sep)
+        path_list[ii] = [part for part in path.split(osp.sep) if part]
+
+    def recurse_level(level_idx):
+        # if toks are all empty we need not have reucrsed here
+        if not any(level_idx.values()):
```

Comment:
Yes that one, sorry I selected the wrong line :fearful: ... 

Also I guess comments look better if they start with a Capital letter, so that if it is a multiline comment you know that is is a paragraph, etc... is style...